presto-iceberg/src/test/java/com/facebook/presto/iceberg/IcebergDistributedTestBase.java

```diff
@@ -1829,6 +1829,24 @@ public void testFilterWithRemainingPredicate(boolean pushdownFilterEnabled)
         assertQuerySucceeds("DROP TABLE test_filterstats_remaining_predicate");
     }
 
+    @Test
+    public void testBatchReadOnTimeType()
+    {
+        assertQuerySucceeds("CREATE TABLE time_batch_read(i time)");
+        assertUpdate("INSERT INTO time_batch_read VALUES time '1:00:00.000'", 1);
+        Session disabledBatchRead = Session.builder(getSession())
+                .setCatalogSessionProperty("iceberg", PARQUET_BATCH_READ_OPTIMIZATION_ENABLED, "false")
+                .build();
+        Session enabledBatchRead = Session.builder(getSession())
+                .setCatalogSessionProperty("iceberg", PARQUET_BATCH_READ_OPTIMIZATION_ENABLED, "true")
+                .build();
+        @Language("SQL") String query = "SELECT * FROM time_batch_read LIMIT 1";
+        MaterializedResult disabledResult = getQueryRunner().execute(disabledBatchRead, query);
+        MaterializedResult enabledResult = getQueryRunner().execute(enabledBatchRead, query);
+        assertEquals(disabledResult, enabledResult);
+        assertQuerySucceeds("DROP TABLE time_batch_read");
```

Comment:
We should cover the scenarios for both dictionary encoding and plain encoding. Besides it's better to test for both `v1` and `v2` parquet version.

Reply:
updated